src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs

```diff
@@ -145,6 +145,32 @@ private void DownloadPackage(object sender, RoutedEventArgs e)
             packageInstaller.DownloadAndInstallPackage(package);
             DependencyRegen(currentWorkspace);
         }
+
+        /// <summary>
+        /// Handler of button which user click when choosing to keep the
+        /// installed version of package instead of the specified one.
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void KeepLocalPackage(object sender, RoutedEventArgs e)
+        {
+            var info = ((PackageDependencyRow)((Button)sender).DataContext).DependencyInfo;
```

Comment:
all the casting makes me nervous - should we be more defensive here? or wrap in a try catch?

Reply:
hmm This private method is directly bind to the button so there should be no surprise here.. But I guess to be safe I can do it..